code/game/objects/items/tcg/tcg.dm

```diff
@@ -0,0 +1,323 @@
+		//Makes a new card based of the series of the pack.
+		new /obj/item/tcgcard(get_turf(user), series, template)
+	to_chat(user, "<span_class='notice'>Wow! Check out these cards!</span>")
+	new /obj/effect/decal/cleanable/wrapping(get_turf(user))
+	playsound(loc, 'sound/items/poster_ripped.ogg', 20, TRUE)
+	if(prob(contains_coin))
+		to_chat(user, "<span_class='notice'>...and it came with a flipper, too!</span>")
+		new /obj/item/coin/thunderdome(get_turf(user))
+	qdel(src)
+
+/obj/item/coin/thunderdome
+	name = "Thunderdome Flipper"
+	desc = "A Thunderdome TCG flipper, for deciding who gets to go first. Also conveniently acts as a counter, for various purposes."
+	icon = 'icons/obj/tcg.dmi'
+	icon_state = "coin_nanotrasen"
+	custom_materials = list(/datum/material/plastic = 400)
+	material_flags = NONE
+	sideslist = list("nanotrasen", "syndicate")
+
+/obj/item/coin/thunderdome/Initialize()
+	. = ..()
+	transform = matrix(0.4,0,0,0,0.4,0)
+
+/obj/item/coin/thunderdome/equipped(mob/user, slot, initial)
+	. = ..()
+	transform = matrix()
+
+/obj/item/coin/thunderdome/dropped(mob/user, silent)
+	. = ..()
+	transform = matrix(0.4,0,0,0,0.4,0)
+
+///Returns a list of cards ids of card_cnt weighted by rarity from the pack's tables that have matching series, with gnt_cnt of the guarenteed table.
+/obj/item/cardpack/proc/buildCardListWithRarity(card_cnt, rarity_cnt)
+	var/list/toReturn = list()
+	//You can always get at least one of some rarity
+	toReturn += returnCardsByRarity(rarity_cnt, guar_rarity)
+	toReturn += returnCardsByRarity(card_cnt, rarity_table)
+	return toReturn
+
+///Returns a list of card datums of the length cardCount that match a random rarity weighted by rarity_table[]
+/obj/item/cardpack/proc/returnCardsByRarity(cardCount, list/rarity_table)
+	var/list/toReturn = list()
+	for(var/card in 1 to cardCount)
+		var/rarity = 0
+		//Some number between 1 and the sum of all values in the list
+		var/weight = 0
+		for(var/chance in rarity_table)
+			weight += rarity_table[chance]
+		var/random = rand(weight)
+		for(var/bracket in rarity_table)
+			//Steals blatently from pickweight(), sorry buddy I need the index
+			random -= rarity_table[bracket]
+			if(random <= 0)
+				rarity = bracket
+				break
+		//What we're doing here is using the cached the results of the rarity we find.
+		//This allows us to only have to run this once per rarity, ever.
+		//Unless you reload the cards of course, in which case we have to do this again.
+		var/list/cards = GLOB.cached_cards[series][rarity]
+		if(cards.len)
+			toReturn += pick(cards)
+		else
+			//If we still don't find anything yell into the void. Lazy coders.
+			log_runtime("The index [rarity] of rarity_table does not exist in the global cache")
+	return toReturn
+
+/datum/card
+	///Unique ID, for use in lookups and (eventually) for persistence. MAKE SURE THIS IS UNIQUE FOR EACH CARD IN AS SERIES, OR THE ENTIRE SYSTEM WILL BREAK, AND I WILL BE VERY DISAPPOINTED.
+	var/id = -1
```

Comment:
My only issue left with the PR is this id.

Reply:
I understand what you mean after reading the closed review. The only time the id value should be touched manually is in the creation of the json files, and in mapping, when placing down a specific card. They are not meant to be manipulated on runtime.

The intent is to allow someone to put a reference to a card datum in the world, or the same for database storage. 

If it wasn't for that at this point I would be using an order variable like you described, but if I do that, things like mapping in cards becomes guesswork, and so does spawning them in a specific case, like a mob's death.

I think I will replace the id with a unique string, I'm sorry for all the faffing about.